x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/cardinality.tsx

```diff
@@ -75,16 +77,14 @@ export const cardinalityOperation: OperationDefinition<CardinalityIndexPatternCo
           : undefined,
     };
   },
-  toEsAggsConfig: (column, columnId) => ({
-    id: columnId,
-    enabled: true,
-    type: OPERATION_TYPE,
-    schema: 'metric',
-    params: {
+  toEsAggsFn: (column, columnId) => {
+    return buildExpressionFunction<AggFunctionsMapping['aggCardinality']>('aggCardinality', {
```

Comment:
This approach looks great to me, seems like that's just as maintainable as the old version 👍

Reply:
> seems like that's just as maintainable as the old version

Yeah after our discussion I realized that, once converting to use the function builder, you gain a lot more type safety than you had previously, so overall I think that will help prevent any future issues 🙂